pkg/codegen/go/gen_program.go

```diff
@@ -32,16 +32,31 @@ type generator struct {
 	scopeTraversalRoots codegen.StringSet
 	arrayHelpers        map[string]*promptToInputArrayHelper
 	isErrAssigned       bool
+	importAliases       map[string]map[string]string
+	importBasePaths     map[string]string
+	modToPkg            map[string]map[string]string
```

Comment:
Instead of separating these maps, I think it might be cleaner to use a single `map[string]*schema.Package`. You could then rephrase some of the helpers in terms of `*schema.Package`, which would make it easier to take advantage of the changes in https://github.com/pulumi/pulumi/pull/5114 as a follow-up change.

Reply:
Oh, good idea! For now, I've just added a TODO for context next time we're working in this area.

Reply:
I'd prefer that we replace the maps now if it's not too much trouble. The bit about taking advantage of #5114 can definitely wait, though.